monai/data/transforms/adaptors.py

```diff
@@ -0,0 +1,207 @@
+
+        if sig.found_kwargs:
+            must_be_types_or_none('inputs', inputs, (dict,))
+            # we just forward all arguments unless we have been provided an input map
+            if inputs is None:
+                dinputs = dict(ditems)
+            else:
+                # dict
+                dinputs = map_names(ditems, inputs)
+
+        else:
+            # no **kwargs
+            # select only items from the method signature
+            dinputs = dict((k, v) for k, v in ditems.items() if k in sig.non_var_parameters)
+            must_be_types_or_none('inputs', inputs, (str, list, tuple, dict))
+            if inputs is None:
+                pass
+            elif isinstance(inputs, str):
+                if len(sig.non_var_parameters) != 1:
+                    raise ValueError("if 'inputs' is a string, function may only have a single non-variadic parameter")
+                dinputs = {inputs: ditems[inputs]}
+            elif isinstance(inputs, (list, tuple)):
+                dinputs = dict((k, dinputs[k]) for k in inputs)
+            else:
+                # dict
+                dinputs = map_only_names(ditems, inputs)
+
+        ret = function(**dinputs)
+
+        # now the mapping back to the output dictionary depends on outputs and what was returned from the function
+        op = outputs
+        if isinstance(ret, dict):
+            must_be_types_or_none('outputs', op, (dict,))
+            if op is not None:
+                ret = {v: ret[k] for k, v in op}
+        elif isinstance(ret, (list, tuple)):
+            if len(ret) == 1:
+                must_be_types('outputs', op, (str, list, tuple))
+            else:
+                must_be_types('outputs', op, (list, tuple))
+
+            if isinstance(op, str):
+                op = [op]
+
+            if len(ret) != len(outputs):
+                raise ValueError("'outputs' must have the same length as the number of elements that were returned")
+
+            ret = dict((k, v) for k, v in zip(op, ret))
+        else:
+            must_be_types('outputs', op, (str, list, tuple))
+            if isinstance(op, (list, tuple)):
+                if len(op) != 1:
+                    raise ValueError("'outputs' must be of length one if it is a list or tuple")
+                op = op[0]
+            ret = {op: ret}
+
+        ditems = dict(ditems)
+        for k, v in ret.items():
+            ditems[k] = v
+
+        return ditems
+
+    return _inner
+
+
+@monai.utils.export('monai.data.transforms')
+class FunctionSignature:
+    def __init__(self, function):
+        import inspect
+        sfn = inspect.signature(function)
+        self.found_args = False
+        self.found_kwargs = False
+        self.defaults = {}
+        self.non_var_parameters = set()
+        for p in sfn.parameters.values():
+            if p.kind is inspect.Parameter.VAR_POSITIONAL:
+                self.found_args = True
+            if p.kind is inspect.Parameter.VAR_KEYWORD:
```

Comment:
line 195 should be `elif` instead of `if`?

Reply:
Nope, it is correct. This code is setting flags if it found that parameter type present

Reply:
if the function input is 
```
def foo(image, *myargs): 
        pass
``` 
`self.non_var_parameters` will have `myargs` perhaps that's not the intended?